src/Service/UserService.php

```diff
@@ -8,10 +8,56 @@
 
 namespace App\Service;
 
+use Doctrine\Common\Persistence\ObjectManager;
 use Doctrine\ORM\EntityManager;
 use App\Entity\User;
 
 class UserService
 {
+    private $manager;
+    private $templating;
+
+    public function __construct(ObjectManager $manager, \Twig_Environment $templating)
```

Comment:
The service should have no notion of twig inside it, that's the job of the controller. The service should simply return data to the controller, who can then use it to render, redirect etc.

Reply:
Yep, I try to move this render in the controller but failed, maybe we can see this together ? (pair prog ?)